src/components/Navbar/Navbar.jsx

```diff
@@ -11,10 +11,20 @@ import {
   VscAccount,
 } from "../utilities/icons";
 import "./Navbar.css";
-import { Link } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
+import { useAuth } from "../../context/auth-context";
+import { useData } from "../../context/data-context";
 
 const Navbar = () => {
   const [menu, setMenu] = useState(false);
+  const navigate = useNavigate();
+  const { token } = useAuth();
+  const { dispatch, search } = useData();
+
+  const searchHandler = (e) => {
+    navigate('/home')
+    dispatch({ type: "SEARCH", payload: e.target.value });
```

Comment:
I will suggest to make the action type as constants as it will reduce further bugs.